Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/command/CommandHandler.java

```diff
@@ -0,0 +1,316 @@
+     *
+     * @param executorObject The object which contains command and completer methods
+     */
+
+    public void register( Object executorObject ) {
+        checkNotNull( executorObject );
+
+        for ( Method method : executorObject.getClass().getMethods() ) {
+            if ( method.isAnnotationPresent( CommandInfo.class ) ) {
+                CommandInfo commandInfo = method.getAnnotation( CommandInfo.class );
+
+                if ( method.getParameterCount() != 1 || !method.getParameterTypes()[0].equals( CommandArguments.class ) ) {
+                    log.warning( "Could not register command " + method.getName() + " in class " + executorObject.getClass().getName() + ": Method may only have a single CommandArguments parameter!" );
+                    continue;
+                }
+
+                registerCommand( commandInfo.name(), commandInfo.description(), commandInfo.usage(), method, executorObject );
+                for ( String alias : commandInfo.aliases() ) {
+                    registerCommand( alias, commandInfo.description(), commandInfo.usage(), method, executorObject );
+                }
+            } else if ( method.isAnnotationPresent( CompleterInfo.class ) ) {
+                CompleterInfo completerInfo = method.getAnnotation( CompleterInfo.class );
+
+                if ( method.getParameterCount() != 1 || !method.getParameterTypes()[0].equals( CommandArguments.class ) ) {
+                    log.warning( "Could not register completer " + method.getName() + " in class " + executorObject.getClass().getName() + ": Method may only have a single CommandArguments parameter!" );
+                    continue;
+                }
+
+                if ( method.getReturnType() != List.class ) {
+                    log.warning( "Could not register completer " + method.getName() + " in class " + executorObject.getClass().getName() + ": Method needs to return a List!" );
+                    continue;
+                }
+
+                registerCompleter( completerInfo.name(), method, executorObject );
+                for ( String alias : completerInfo.aliases() ) {
+                    registerCompleter( alias, method, executorObject );
+                }
+            }
+        }
+    }
+
+    /**
+     * Registers the registered commands into the bukkit help map
+     */
+    public void registerHelp() {
+        Set<HelpTopic> help = new TreeSet<>( HelpTopicComparator.helpTopicComparatorInstance() );
+        for ( String commandLabel : commandMap.keySet() ) {
+            if ( !commandLabel.contains( "." ) ) {
+                org.bukkit.command.Command cmd = bukkitCommandMap.getCommand( commandLabel );
+                HelpTopic topic = new GenericCommandHelpTopic( cmd );
+                help.add( topic );
+            }
+        }
+        //TODO which perm node is requiered to be able to see the help? Where do we get that perm node from? Hardcode?
```

Comment:
I would suggest <system>.command.<commandname>.help. For example raid.command.start.help for "/raid start" help